public/codingprogress.js

```diff
@@ -0,0 +1,113 @@
+class TableController {
+    static updateCodingProgressTable(codingProgressData) {
+        let data = codingProgressData.data,
+            lastUpdate = codingProgressData.lastUpdate; 
+
+        // Set last updated timestamp in UI
+        document.getElementById("last-update").innerText = `Last updated: ${lastUpdate}`;
+
+        // Check the state of the column sorted 
+        if (!TableController.column) 
+            TableController.column = "Done";
+
+        // Save sorting information
+        let sortInfo = { column: TableController.column, order: "" };
+
+        // Invoke `transform` function with column to be sorted on page load
+        transform(sortInfo.column);
+        
+        // Function used to generate coding progress table
+        function transform(column) {
+            // Toggle sorting state
+            if (sortInfo.order === "descending" && column === sortInfo.column)
+                sortInfo.order = "ascending";
+            else { sortInfo.order = "descending"; sortInfo.column = column }
+
+            // Keep the state of the column sorted to avoid its reset on update
+            TableController.column = sortInfo.column;
+
+            d3.select("tbody").selectAll("tr").remove();
+
+            // Table Header
+            d3.select("thead").selectAll("th")
+                .data(TableController.jsonToArray(data[0]))
+                .enter().append("th")
+                .attr("class", "table-heading")
+                .on("click", (d, i, n) => transform(d[0]))
```

Comment:
```suggestion
                .on("click", (d) => transform(d[0]))
```
Then I don't have to think about what irrelevant args mean